packages/react-aria-components/src/GridList.tsx

```diff
@@ -99,37 +254,133 @@ function GridListItem({item}) {
       isHovered,
       isFocusVisible,
       selectionMode: state.selectionManager.selectionMode,
-      selectionBehavior: state.selectionManager.selectionBehavior
+      selectionBehavior: state.selectionManager.selectionBehavior,
+      allowsDragging: !!dragState,
+      isDragging,
+      isDropTarget: dropIndicator?.isDropTarget
+    }
+  });
+
+  let renderDropIndicator = dragAndDropHooks?.renderDropIndicator || (target => <DropIndicator target={target} />);
+  let dragButtonRef = useRef(null);
+  useEffect(() => {
+    if (dragState && !dragButtonRef.current) {
+      console.warn('Draggable items in a GridList must contain a <Button slot="drag"> element so that keyboard and screen reader users can drag them.');
+    }
+  // eslint-disable-next-line
+  }, []);
+
+  return (
+    <>
+      {dragAndDropHooks?.useDropIndicator &&
+        renderDropIndicator({type: 'item', key: item.key, dropPosition: 'before'})
+      }
+      {dropIndicator && !dropIndicator.isHidden &&
+        <div role="row">
+          <div role="gridcell">
+            <div role="button" {...visuallyHiddenProps} {...dropIndicator?.dropIndicatorProps} ref={dropIndicatorRef} />
+          </div>
+        </div>
+      }
```

Comment:
I believe it was because the plain text sibling in the cell would become a swipable target on iOS since there is no wrapping element to add `aria-hidden` to. Putting it in a separate element means the original row can be completely hidden.